framework/src/test/java/org/tron/common/utils/PublicMethod.java

```diff
@@ -105,11 +109,30 @@ public static GrpcAPI.Return broadcastTransaction(
     return response;
   }
 
+  public static int chooseRandomPort() {
+    return chooseRandomPort(10240, 65000);
+  }
+
   public static int chooseRandomPort(int min, int max) {
-    return new Random().nextInt(max - min + 1) + min;
+    int port = new Random().nextInt(max - min + 1) + min;
+    try {
+      while (!checkPortAvailable(port)) {
+        port = new Random().nextInt(max - min + 1) + min;
+      }
+    } catch (IOException e) {
+      return new Random().nextInt(max - min + 1) + min;
```

Comment:
Why should get a new random number when an IOException catched? Does it means the port is available if an IOException catched?

Reply:
This method just picks a random port, if the check logic can not return a value, we should return an alternative port.

Reply:
So shall we check the alternative port again?

Reply:
I think it will be too complicated, if the probability of picking a useless port is 1%, double check will decrease the probability to 0.01%, most of time it will work well